pwnlib/elf/elf.py

```diff
@@ -211,16 +225,24 @@ class ELF(ELFFile):
     _fill_gaps = True
 
 
-    def __init__(self, path, checksec=True):
+    def __init__(self, path_or_bytes, checksec=True):
         # elftools uses the backing file for all reads and writes
         # in order to permit writing without being able to write to disk,
         # mmap() the file.
 
-        #: :class:`file`: Open handle to the ELF file on disk
-        self.file = open(path,'rb')
-
-        #: :class:`mmap.mmap`: Memory-mapped copy of the ELF file on disk
-        self.mmap = mmap.mmap(self.file.fileno(), 0, access=mmap.ACCESS_COPY)
+        if isinstance(path_or_bytes, (bytes, bytearray)) and path_or_bytes.startswith(b'\x7FELF'):
+            # `self.file.name` is used for error reporting, so we need to use a custom wrapper class
+            # to allow setting that field
+            self.file = self.mmap = NamedMmap(-1, len(path_or_bytes))
+            self.file.name = '<anonymous>'
+            self.mmap.write(path_or_bytes)
+            path = ""
```

Comment:
Also, I'm not 100% sure on the best option for what path should be (dummy string vs empty string vs `None` vs not defined)

Reply:
I think `None` maybe? Or simply nothing, I also think it might be worth it to just use `getattr(self.file, 'name', None)` at the usage place instead of creating a custom class just for adding an extra field.

Also, we already have [self.path](https://docs.pwntools.com/en/latest/elf/elf.html#pwnlib.elf.elf.ELF.path) just for that. I believe we can use self.path safely in that case and skip the extra gymnastics. So just changing corefile logs to use self.path seems best.